qemu/tests/ioeventfd.py

```diff
@@ -0,0 +1,166 @@
+
+    def _common_io_test():
+        """ Execute common io testing inside guest. """
+        logging.info('Doing io testing inside guest.')
+        op_cmd = params['op_cmd']
+        if os_type == 'windows':
+            op_cmd = utils_misc.set_winutils_letter(session, op_cmd)
+        o = session.cmd(op_cmd, float(params['op_timeout']))
+        logging.debug(o)
+
+    def _iozone_test(session):
+        """ Execute iozone testing inside guest. """
+        logging.info('Doing iozone inside guest.')
+        if os_type == 'windows':
+            if params['drive_format'] == 'scsi-hd':
+                session = utils_test.qemu.windrv_check_running_verifier(
+                    session, vm, test, params["driver_name"])
+        try:
+            iozone = generate_instance(params, session, 'iozone')
+            iozone.run(params['iozone_options'], float(params['iozone_timeout']))
+        finally:
+            iozone.clean()
+            return session
+
+    def _check_property(vm, ioeventfd_opt):
+        """
+        Check the value of ioeventfd by sending "info qtree" in QMP.
+        """
+        ioevent_qtree_val = 'true' if 'on' in ioeventfd_opt else 'false'
+        logging.info('Execute info qtree in QMP monitor.')
+        qtree = qemu_qtree.QtreeContainer()
+        qtree.parse_info_qtree(vm.monitor.human_monitor_cmd("info qtree"))
+        for node in qtree.get_nodes():
+            if isinstance(node, qemu_qtree.QtreeDev) and (
+                    node.qtree.get('id', None) == dev_id):
+                if node.qtree.get('ioeventfd', None) is None:
+                    test.fail('The qtree device %s has no property ioeventfd.'
+                              % dev_id)
+                elif node.qtree['ioeventfd'] == ioevent_qtree_val:
+                    logging.info(
+                        'The \"%s\" matches with qtree device \"%s\"(%s).' %
+                        (ioeventfd_opt, dev_id, ioevent_qtree_val))
+                    break
+                else:
+                    test.fail(
+                        'The \"%s\" mismatches with qtree device \"%s\"(%s).' %
+                        (ioeventfd_opt, dev_id, ioevent_qtree_val))
+        else:
+            test.error('No such \"%s\" qtree device.' % dev_id)
+
+    def _get_ioeventfds(ioeventfd_opt):
+        """
+        Get the number of ioeventfds insid host.
+        """
+        logging.info('Check the \"%s\" via /proc/$PID/fd/.' % ioeventfd)
+        dst_log = 'off' if 'off' in ioeventfd_opt else 'on'
+        cmd = 'ls -l /proc/$(pgrep qemu-kvm)/fd > /tmp/%s' % dst_log
+        logging.debug('Running \'%s\'' % cmd)
+        s, o = process.getstatusoutput(cmd)
+        if s:
+            test.error('Failed to get the number of event fd.\n%s' % o)
+
+    def _compare_ioeventfds():
+        """
+        Compare the number of ioeventfds between "off" and "on".
+        """
+        logging.info('Compare the output of \'ls -l /proc/$PID/fd/\'.')
```

Comment:
Please update :
Compare fd number of ioeventfd=on between ioeventfd=off

Reply:
thanks, will update